Introductory Tests/Trajectory.py

```diff
@@ -0,0 +1,170 @@
+            algorithm (int): User defined algorithm of choice
+            radius_bounding_sphere (float)_: Radius for the bounding sphere around mesh.
+        """
+        # Creating the mesh (TetGen)
+        self.body_mesh, self.mesh_vertices, self.mesh_faces, largest_body_protuberant = mesh_utility.create_mesh()
+
+        # Assertions:
+        assert body_density > 0
+        assert final_time > start_time
+        assert time_step <= (final_time - start_time)
+        assert radius_bounding_sphere > largest_body_protuberant
+
+        # Setup equations of motion class
+        self.eq_of_motion = EquationsOfMotion(self.mesh_vertices, self.mesh_faces, body_density)
+
+        # Additional hyperparameters
+        self.start_time = start_time
+        self.final_time = final_time
+        self.time_step = time_step
+        self.algorithm = algorithm
+        self.radius_bounding_sphere = radius_bounding_sphere
+
+
+    def integrate(self, x: np.ndarray) -> Union[np.ndarray, float, float]:
+        """compute_trajectory computes trajectory of satellite using numerical integation techniques 
+
+        Args:
+            x (np.ndarray): State vector containing values for position and velocity of satelite in three dimensions.
+
+        Returns:
+            trajectory_info (np.ndarray): Numpy array containing information on position and velocity at every time step (columnwise).
+            squared_altitudes (float): Sum of squared altitudes above origin for every position
+            collision_penalty (float): Penalty value given for the event of a collision with the celestial body.
+        """
+
+        # Integrate trajectory
+        initial_state = D.array(x)
+        trajectory = de.OdeSystem(
+            self.eq_of_motion.compute_motion, 
+            y0 = initial_state, 
+            dense_output = True, 
+            t = (self.start_time, self.final_time), 
+            dt = self.time_step, 
+            rtol = 1e-12, 
+            atol = 1e-12,
+            constants=dict(risk_zone_radius = self.radius_bounding_sphere)) #, mesh_vertices = self.mesh_vertices, mesh_faces = self.mesh_faces
+        trajectory.method = str(IntegrationScheme(self.algorithm).name)
+
+        point_is_inside_risk_zone.is_terminal = False
+        trajectory.integrate(events=point_is_inside_risk_zone)
+        trajectory_info = np.vstack((np.transpose(trajectory.y), trajectory.t))
+
+        # Compute average distance to target altitude
+        squared_altitudes = trajectory_info[0,:]**2 + trajectory_info[1,:]**2 + trajectory_info[2,:]**2
+
+        # Add collision penalty
+        points_inisde_risk_zone = np.empty((len(trajectory.events), 3), dtype=np.float64)
+        i = 0
+        for j in trajectory.events:
+            points_inisde_risk_zone[i,:] = j.y[0:3]
+            i += 1
+        
+        collision_avoided = point_is_outside_mesh(points_inisde_risk_zone, self.mesh_vertices, self.mesh_faces)
+        if all(collision_avoided) == True:
```

Comment:
Personally I would return just that a collision occurred  and decide penalty in UDP. The fitness value should not be part of this I think?

Reply:
I will move it to the UDP. 

When I coded this part, my idea was to mainly add a very large weight to the objective function whenever there was a collision somewhere along the trajectory, and in that way mark the given chromosome as infeasible. Are there any other ways of doing something similar in pygmo without including the fitness function?

Reply:
I think your idea is correct, it's just that it should happen in the UDP from a conceptual point because the UDP is responsible for deciding fitness while this here is responsible for trajectory / collisions

Reply:
That makes sense, I will move it!